src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -100,8 +100,14 @@ public static void initialize(ModelDao modelDao) {
     }
 
     private static float[] ObjectsToFloats(List<Object> objs) {
+        if (Objects.isNull(objs)) {
+            throw new IllegalArgumentException("[" + NAME + "] requires 'vector' to be non-null");
```

Comment:
In case of parametrized error messages I think we're using `String.format()`

Reply:
Updated